File: java/src/main/java/io/fivexer/sdk/model/TeamMembers.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A team's roster, as returned by the members endpoint.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TeamMembers {
    private String teamId;
    private List<TeamMember> members;

    public String getTeamId() { return teamId; }
    public List<TeamMember> getMembers() { return members; }
}

File: php/src/Model/TeamMembers.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A team's roster.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TeamMembers
{
    public function __construct(
        public readonly string $teamId,
        /** @var list<TeamMember> */
        public readonly array $members,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            teamId: (string) ($data['teamId'] ?? ''),
            members: Json::parseEach($data, 'members', [TeamMember::class, 'fromArray']),
        );
    }
}
